File: Balanced_Binary_Tree.java
class Balanced_Binary_Tree {
    boolean isBal = true;
    public boolean isBalanced(TreeNode root) {
        maxDepth(root);
        return isBal;
        
    }
    public int maxDepth(TreeNode node){
        if(node == null) return 0 ;
        int leftSub = maxDepth(node.left);
        int rightSub = maxDepth(node.right);
        if(Math.abs(leftSub-rightSub)>1)
            isBal = false;

        return Math.max(leftSub, rightSub)+1;
    }
}

File: BalancedBinaryTree.kt
class BalancedBinaryTree {
    fun isBalanced(root: TreeNode?): Boolean {
        if (root == null) return true
        val left = height(root.left)
        val right = height(root.right)
        if (kotlin.math.abs(left - right) > 1) return false
        return isBalanced(root.left) && isBalanced(root.right)
    }

    fun height(root: TreeNode?): Int {
        if (root == null) return 0
        return maxOf(height(root.left), height(root.right)) + 1
    }
}
